include/triton/Conversion/TritonGPUToLLVM/Utility.h

```diff
@@ -866,9 +866,15 @@ inline void emitWmmaOffsetForCTA(const AMDWmmaEncodingAttr &wmmaLayout,
   if (rank == 3)
     elemOffset[0] = ctaBatchOffset;
   for (unsigned elem = 0; elem < elemsPerThreadPerGroup; elem++) {
-    elemOffset[rank - 2] =
-        ctaOffsetX * shapePerCta[rank - 2] + elemStride * elem;
-    elemOffset[rank - 1] = ctaOffsetY * shapePerCta[rank - 1];
+    if (wmmaLayout.getIsTransposed()) {
+      elemOffset[rank - 1] =
+          ctaOffsetX * shapePerCta[rank - 1] + elemStride * elem;
+      elemOffset[rank - 2] = ctaOffsetY * shapePerCta[rank - 2];
+    } else {
+      elemOffset[rank - 2] =
+          ctaOffsetX * shapePerCta[rank - 2] + elemStride * elem;
+      elemOffset[rank - 1] = ctaOffsetY * shapePerCta[rank - 1];
```

Comment:
Do we really need this code? As I understand it is required as alternative for LL conversion, but LL convert should work fine. As well as https://github.com/triton-lang/triton/pull/5301/files#diff-ea09e37788131c30733ddee48bf7871b744be17b233c5650f679e3e2b8f7c7eeR928-R940

Reply:
We can switch to use this code by setting `useLegacyMMAConversion` variable to true: https://github.com/triton-lang/triton/blob/main/lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM.cpp#L29

I believe I've tried it some time ago, but did not try it with current revision.
I will verify if it works as soon as I get access to hardware.